test/integration/YieldTokenIncidents/submitClaim.js

```diff
@@ -17,59 +18,78 @@ const setTime = async timestamp => {
-  it('submits DAI claim and approves claim', async function () {
-    const { DEFAULT_PRODUCT_INITIALIZATION } = this;
-    const { cover, stakingPool0, as, tk, dai, yc } = this.contracts;
-    const [coverBuyer1, staker1, staker2, member1] = this.accounts.members;
-    const [nonMember1, nonMember2] = this.accounts.nonMembers;
-
-    const productId = 0;
-    const payoutAsset = 1; // DAI
-    const period = 3600 * 24 * 30; // 30 days
+describe('submitClaim', function () {
+  beforeEach(async function () {
+    const { tk } = this.contracts;
 
-    const amount = parseEther('1');
+    const members = this.accounts.members.slice(0, 5);
+    const amount = parseEther('10000');
+    for (const member of members) {
+      await tk.connect(this.accounts.defaultSender).transfer(member.address, amount);
+    }
+  });
 
-    const stakingAmount = parseEther('100');
-    await tk.connect(this.accounts.defaultSender).transfer(staker1.address, stakingAmount);
-    await tk.connect(this.accounts.defaultSender).transfer(staker2.address, stakingAmount);
+  function calculateFirstTrancheId(lastBlock, period, gracePeriod) {
+    return Math.floor((lastBlock.timestamp + period + gracePeriod) / (91 * 24 * 3600));
+  }
 
+  async function stake({ stakingPool, staker, productId, period, gracePeriod }) {
+    // Staking inputs
+    const stakingAmount = parseEther('6000');
     const lastBlock = await ethers.provider.getBlock('latest');
+    const firstTrancheId = calculateFirstTrancheId(lastBlock, period, gracePeriod);
 
-    const firstTrancheId = Math.floor(lastBlock.timestamp / (91 * 24 * 3600));
-
-    await stakingPool0.connect(staker1).depositTo([
+    // Stake to open up capacity
+    await stakingPool.connect(staker).depositTo([
       {
         amount: stakingAmount,
         trancheId: firstTrancheId,
         tokenId: 1, // new position
-        destination: ZERO_ADDRESS,
+        destination: AddressZero,
       },
     ]);
+    await stakingPool.setTargetWeight(productId, 10);
+  }
+
+  it('submits ETH claim and approves claim', async function () {
+    const { DEFAULT_PRODUCT_INITIALIZATION } = this;
+    const { cover, stakingPool0, as, yc, ybETH } = this.contracts;
+    const [coverBuyer1, staker1] = this.accounts.members;
+    const [nonMember1] = this.accounts.nonMembers;
+
+    const productId = 2; // ybETH
+    const coverAsset = 0; // ETH
+    const period = 3600 * 24 * 30; // 30 days
+    const gracePeriod = 3600 * 24 * 30;
+
+    const amount = parseEther('10');
+
+    // Stake to open up capacity
+    await stake({ stakingPool: stakingPool0, staker: staker1, productId, period, gracePeriod });
 
+    // Buy Cover
     const expectedPremium = amount
       .mul(BigNumber.from(DEFAULT_PRODUCT_INITIALIZATION[0].targetPrice))
       .div(BigNumber.from(priceDenominator));
 
-    await stakingPool0.setTargetWeight(productId, 10);
+    const expectedMinCoveredAmount = amount.sub(COVERED_AMOUNT_PRECISION_ERROR_TOLERANCE);
```

Comment:
Is this something undeterministic that we cannot get precisely in js?

Reply:
It's a precision error range. the precision error is dust, just 20 wei